parse.go

```diff
@@ -55,9 +55,17 @@ var (
 	ErrorMismatchPixelDataLength = errors.New("the size calculated from DICOM elements and the PixelData element's VL are mismatched")
 )
 
+func Parse(in io.Reader, bytesToRead int64, frameChan chan *frame.Frame, opts ...ParseOption) (Dataset, error) {
+	return parseInternal(in, bytesToRead, frameChan, opts...)
+}
+
+func ParseUntilEOF(in io.Reader, frameChan chan *frame.Frame, opts ...ParseOption) (Dataset, error) {
```

Comment:
(no specific action, just a discussion thought): Would it make sense to have this be an option passed the Parse() as is? e.g. you'd pass a nonsense bytesToRead, but with a dicom.ParseUntilEOFOption. I don't know that I love that, but just a thought to throw out there. 

Eventually if this works well, we'll prob hard migrate the Parse API to not have bytesToRead by default, but for folks who want a strict bytesToRead, they can pass that in as an option (this may be useful to detect other issues like what if the whole dicom isn't read before encountering an EOF for some reason).

Reply:
I do think that we should have the main function just parse, without forcing people to provide the bytesToRead (which is counter intuitive in most cases) that's why I created the ParseUntilEOF function.
I just feel that if we don't hard migrate right now, the Parse function with a "dicom.ParseUntilEOFOption" option will not be visible enough for people to adopt it...
**I'm in favour of the breaking change, but I couldn't envision pitching it to you outright :)**

Reply:
Yeah, it is fair that the dicom.ParseOption may not be as "visible" though we could update some docstrings in the Parse Godoc to call it out. 

I guess for now, let's leave it as it is (as a separate function), and then over time we can migrate it over to being the main API when we're ready for it (potentially doing it together with another breaking change I may want to make).